src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/IDE/VSSolutionManager.cs

```diff
@@ -37,7 +37,7 @@ namespace NuGet.PackageManagement.VisualStudio
     [Export(typeof(ISolutionManager))]
     [Export(typeof(IVsSolutionManager))]
     [PartCreationPolicy(CreationPolicy.Shared)]
-    public sealed class VSSolutionManager : IVsSolutionManager, IVsSelectionEvents, IVsSolutionEvents, IDisposable
+    public sealed class VSSolutionManager : IVsSolutionManager, IVsSelectionEvents, IVsSolutionEvents, IVsSolutionLoadEvents, IDisposable
```

Comment:
Was this new `IVsSolutionLoadEvents` necessary? I'm just curious.

Reply:
It's the only way I could see to get a "Begin Solution Load" event. IVsSolutionEvents only has "OnAfterSolutionLoad". It's also a convenient way to get the solution file/directory during solution load.